app/services/journeys/params_validator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Journeys
+  class ParamsValidator
+    include ActiveModel::Validations
+
+    attr_reader :action_name, :timestamp, :billable, :vehicle, :from_location_id, :to_location_id
+
+    validates :billable, inclusion: { in: [true, false] }, allow_nil: true, on: :update # NB: billable is optional on update
```

Comment:
It's not simpler just to set false (or true) as default?
`nil` for boolean is something we should whenever possible

Reply:
hmm... in this case no, billable has a very real financial meaning, if its `false` the supplier will pay and if its `true` the MoJ will pay. So I want it to be explicitly set by the client and not to have an assumed/default value

Reply:
I'd agree, this is a little strange. Booleans should only have two states, not three. If this needs three states then it probably makes more sense to use a different type.

Reply:
the `nil` billable state in the validator in the `update` context means it is not provided, which means "don't update the billable flag"

the logic is:
* when creating a journey, `billable` *must* always be provided (only true or false)
* when updating a journey, `billable` *may* be provided. If it is provided it must only be true or false. If it is not provided, the validator will see its value as nil and ignore it.

The journeys controller will never allow a journey to be created or updated with billable=nil.

Reply:
Okay, that's a good explanation - thanks this makes sense now 👍